src/Backends/IF97/IF97Backend.h

```diff
@@ -59,31 +97,115 @@ class IF97Backend : public AbstractState  {
     // ************************************************************************* //
     //                   Basic Thermodynamic Functions                           //
     // ************************************************************************* //
-	/// Return the mass density in kg/m^3
+    //
+    double calc_Liquid(parameters iCalc) {
+        switch(iCalc){
+        case iDmass:  return IF97::rholiq_p(_p); break; ///< Mass-based density
+        case iHmass:  return IF97::hliq_p(_p); break;   ///< Mass-based enthalpy
+        case iSmass:  return IF97::sliq_p(_p); break;   ///< Mass-based entropy
+        case iCpmass: return IF97::cpliq_p(_p); break;  ///< Mass-based constant-pressure specific heat
+        case iCvmass: return IF97::cvliq_p(_p); break;  ///< Mass-based constant-volume specific heat
+        case iUmass:  return IF97::uliq_p(_p); break;   ///< Mass-based internal energy
+        case ispeed_sound: return IF97::speed_soundliq_p(_p); break;  ///< Speed of Sound
+        default: return -_HUGE;
+        };
+    }
+    double calc_Vapor(parameters iCalc) {
+        switch(iCalc){
+        case iDmass:  return IF97::rhovap_p(_p); break; ///< Mass-based density
+        case iHmass:  return IF97::hvap_p(_p); break;   ///< Mass-based enthalpy
+        case iSmass:  return IF97::svap_p(_p); break;   ///< Mass-based entropy
+        case iCpmass: return IF97::cpvap_p(_p); break;  ///< Mass-based constant-pressure specific heat
+        case iCvmass: return IF97::cvvap_p(_p); break;  ///< Mass-based constant-volume specific heat
+        case iUmass:  return IF97::uvap_p(_p); break;   ///< Mass-based internal energy
+        case ispeed_sound: return IF97::speed_soundvap_p(_p); break;  ///< Speed of Sound
+        default: return -_HUGE;
+        };
+    }
+    double calc_Flash(parameters iCalc) {
+        switch(_phase){
+            case iphase_liquid: 
+                return calc_Liquid(iCalc); 
+                break;
+            case iphase_gas:    
+                return calc_Vapor(iCalc); 
+                break;
+            case iphase_twophase:
+                switch(iCalc){ 
+                case iDmass:
+                    return 1.0/(_Q/calc_Vapor(iCalc) + (1.0-_Q)/calc_Liquid(iCalc)); break;
+                case iCpmass:
+                    throw NotImplementedError(format("Isobaric Specific Heat not implemented in two phase region")); break;
```

Comment:
Yes, those two cases are handled just above that logic for (_phase == iphase_liquid) and (_phase == iphase_vapor), where I only want to call one function instead of two to save time in the spirit of IF97.  This instance is only called when Q is _between_ 0 and 1 (_phase == iphase_two_phase); in which case these quantities (Cp, Cv, Speed of Sound) are undefined.  (Ignore my external e-mail on this, by the way.)